### ISSUE TITLE
Namespace collision with libcombine?

### ISSUE THREAD

**Issue by FFroehlich** [sbmlteam/python-libsbml#23]
A user reported the following issue with AMICI: https://github.com/AMICI-dev/AMICI/issues/1483

What ends up happening is that `sbml_model.all_elements` has type `libcombine.SBaseList` instead of the usual `libsbml.ListWrapperSBase`. 

I can reproduce that this always happens if both libcombine and libsbml are imported simultaneously. It appears that all kinds of other model elements are also stripped of their usual types and appear as `SBase` elements. I suspect that this is some kind of incompatibility that arises from the fact that both are swig exported interfaces.

**Comment by fbergmann** [comment]
indeed, libCombine should probably not be exposing the duplicated symbol (as it won't make any use of that). Until i release a new version i'd recommend to first include libcombine, and then libsedml.

**Comment by leonplanken** [comment]
We ran into this problem as well: a bug that only appeared when importing tellurium.  Thanks to the comments above, I've reduced it to the following minimal example:

```
bug = True
workaround = False

if bug:
    import libsbml
    import libcombine
else:
    import libcombine
    import libsbml

plus_minus = {libsbml.AST_PLUS, libsbml.AST_MINUS}

m = libsbml.parseL3Formula('42')
if workaround:
    m = libsbml.ASTNode(m)
t = m.getType()  # This yields the memory leak message
try:
    if t in plus_minus:  # This yields the TypeError
        pass
except TypeError as e:
    print(e)
else:
    print("don't panic!")
```

A workaround for using the "wrong" import order in the above code is to use the ASTNode copy constructor.

**Comment by fbergmann** [comment]
I'm afraid I still don't have a solution for this, several libraries (sbml, numl, combine, sedml, and sbng) use SWIG generated types out of the core libsbml library. Other than using a custom import order I have no solution for the moment. I'm happy to hear if anyone else found one.

**Comment by leonplanken** [comment]
Thanks for following up; isn't this (also) a SWIG bug?
I was mistaken in my report above, I wrote there was still a memory leak when importing libcombine first (setting type_of_bug="ONLY_MEMORY_LEAK" in my example script) but that's not the case.  I'll edit it out.
